google/oauth2/_client.py

```diff
@@ -257,3 +257,38 @@ def refresh_grant(
     expiry = _parse_expiry(response_data)
 
     return access_token, refresh_token, expiry, response_data
+
+
+'''
+TODO(anibadde)
```

Comment:
If you plan to do this as part of this PR, likely fine, but if you want to come back to this (if that is doable), we should likely open an issue to track the work.

Reply:
Resolved - had logic for async credential refreshing, but this doesn't seem to add any benefit so removed from implementation.